docs/Developer-guide.md

```diff
@@ -0,0 +1,131 @@
+# Developer guide
+
+The following document details the proposed structure for this repo, the benchmark scripts and libraries.
+
+## General design and motivations
+
+The goal of the benchmark repo is to provide scripts to compare tree-based machine learning framework on similar tasks and environment.
+
+The benchmark is made of a collection of scripts. These scripts can be run locally, or manually from within a remote VM. But these scripts will eventually be submitted as a pipeline or workflow chaining scripts in a sequence: data generation, data processing, training, inferencing.
+
+In order to do that, we need each step for benchmarking a given framework to be implemented as a single script with clear inputs, outputs and parameters. We organize those into sub-directories under `/src/scripts/`.
+
+For the benchmark to automate the analysis of the results and the production of a benchmark report, we need each script to report a common set of metrics or tags.
+
+## Proposed structure of the repo
+
+| Directory | Content |
+| :-- | :-- |
+| `/.github/workflows/` | [github workflows](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions) for this project |
+| `/docs/` | Documentation in markdown |
```

Comment:
Bonus points if you use `mkdocs` or something similar to push this to github.io.

Reply:
Thanks for the suggestion, created a request #14